libraries/launcher/org/prismlauncher/log4jinjector/UniversalLog4jInjector.java

```diff
@@ -0,0 +1,38 @@
+package org.prismlauncher.log4jinjector;
+
+import org.prismlauncher.utils.logging.Log;
+
+import java.util.Objects;
+
+public class UniversalLog4jInjector {
+    private static final String EXCEPTION_DURING_INJECTION = "Unable to inject into Log4j. Token may be written to a file as a result";
+    public static void inject() {
+        if (Objects.equals(System.getProperty("org.prismlauncher.log4j.inject"), "false")) {
+            return;
+        }
+        String l4jVersion;
+        try {
+            l4jVersion = Class.forName("org.apache.logging.log4j.LogManager").getPackage().getImplementationVersion();
+            assert l4jVersion != null;
+        } catch (Exception e) {
+            return;
+        }
+        if (l4jVersion.startsWith("2.0")) {
```

Comment:
I would remove the assert and do this instead. I'm not too familiar with this Java keyword, but I'm fairly sure it gets disabled. Additionally it would throw an exception which would be immediately ignored.
```suggestion
        if (l4jVersion == null)
            return;
 
        if (l4jVersion.startsWith("2.0")) {
```